src/System.Linq/src/System/Linq/Where.cs

```diff
@@ -136,17 +138,33 @@ public override IEnumerable<TResult> Select<TResult>(Func<TSource, TResult> sele
                 return new WhereSelectEnumerableIterator<TSource, TResult>(_source, _predicate, selector);
             }
 
+            public TSource[] ToArray() => EnumerableHelpers.ToArray(this);
```

Comment:
This may be slightly penalizing `ToArray` since we're adding an extra virtual call to this method... will post perf tests when I get a chance.